sqlserver/datadog_checks/sqlserver/statements.py

```diff
@@ -59,10 +60,8 @@
 """
 
 PLAN_LOOKUP_QUERY = """\
-select cast(query_plan as nvarchar(max)) as query_plan from sys.dm_exec_query_stats
-    cross apply sys.dm_exec_query_plan(plan_handle)
-where
-    query_hash = CONVERT(varbinary(max), ?, 1) and query_plan_hash = CONVERT(varbinary(max), ?, 1)
+select cast(query_plan as nvarchar(max)) as query_plan
```

Comment:
Query plan is an "XML" data type and some of the drivers we support don't support it, throwing an exception when you're reading it out.

Reply:
Can we selectively do this cast only on drivers that don't support it?

Reply:
Yeah we could do that as a follow-up. Curious how much of an impact it would make. The XML has to be serialized to some format anyway, and I guess the binary format is more compact so it may help a bit.